modules/k8s/daemonset_test.go

```diff
@@ -0,0 +1,105 @@
+package k8s
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetDaemonSetEReturnsErrorForNonExistantDaemonSet(t *testing.T) {
+	t.Parallel()
+
+	options := NewKubectlOptions("", "")
+	_, err := GetDaemonSetE(t, options, "sample-ds")
+	require.Error(t, err)
+}
+
+func TestGetDaemonSetEReturnsCorrectServiceInCorrectNamespace(t *testing.T) {
+	t.Parallel()
+
+	uniqueID := strings.ToLower(random.UniqueId())
+	options := NewKubectlOptions("", "")
+	options.Namespace = uniqueID
+	configData := fmt.Sprintf(EXAMPLE_DAEMONSET_YAML_TEMPLATE, uniqueID, uniqueID)
+	KubectlApplyFromString(t, options, configData)
+	defer KubectlDeleteFromString(t, options, configData)
+
+	daemonSet := GetDaemonSet(t, options, "fluentd-elasticsearch")
+	require.Equal(t, daemonSet.Name, "fluentd-elasticsearch")
+	require.Equal(t, daemonSet.Namespace, uniqueID)
+}
+
+func TestListDaemonSetsReturnsCorrectServiceInCorrectNamespace(t *testing.T) {
+	t.Parallel()
+
+	uniqueID := strings.ToLower(random.UniqueId())
+	options := NewKubectlOptions("", "")
+	options.Namespace = uniqueID
+	configData := fmt.Sprintf(EXAMPLE_DAEMONSET_YAML_TEMPLATE, uniqueID, uniqueID)
+	KubectlApplyFromString(t, options, configData)
+	defer KubectlDeleteFromString(t, options, configData)
+
+	daemonSets := ListDaemonSets(t, options, metav1.ListOptions{})
+	require.Equal(t, len(daemonSets), 1)
+
+	daemonSet := daemonSets[0]
+	require.Equal(t, daemonSet.Name, "fluentd-elasticsearch")
+	require.Equal(t, daemonSet.Namespace, uniqueID)
+}
+
+const EXAMPLE_DAEMONSET_YAML_TEMPLATE = `---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: %s
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: fluentd-elasticsearch
+  namespace: %s
+  labels:
+    k8s-app: fluentd-logging
+spec:
+  selector:
+    matchLabels:
+      name: fluentd-elasticsearch
+  template:
+    metadata:
+      labels:
+        name: fluentd-elasticsearch
+    spec:
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
+      containers:
+      - name: fluentd-elasticsearch
+        image: gcr.io/fluentd-elasticsearch/fluentd:v2.5.1
```

Comment:
Can you use a different, simpler image that doesn't have external dependencies for the unit test? That way, it won't fail on boot up and the config can be simplified (e.g we don't need all the volumes).

E.g a simple alpine container with `sleep 999999`.